lightning/src/ln/channelmanager.rs

```diff
@@ -4993,7 +5006,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 							self.claim_funds_internal(self.channel_state.lock().unwrap(), htlc_update.source, preimage, htlc_update.htlc_value_satoshis.map(|v| v * 1000), true, funding_outpoint.to_channel_id());
 						} else {
 							log_trace!(self.logger, "Failing HTLC with hash {} from our monitor", log_bytes!(htlc_update.payment_hash.0));
-							self.fail_htlc_backwards_internal(self.channel_state.lock().unwrap(), htlc_update.source, &htlc_update.payment_hash, HTLCFailReason::Reason { failure_code: 0x4000 | 8, data: Vec::new() });
+							self.fail_htlc_backwards_internal(self.channel_state.lock().unwrap(), htlc_update.source, &htlc_update.payment_hash, HTLCFailReason::Reason { failure_code: 0x4000 | 8, data: Vec::new() }, HTLCDestination::Payment { payment_hash: htlc_update.payment_hash });
```

Comment:
makes sense, good catch 😅

Reply:
@TheBlueMatt I'm getting a little stuck here with retrieving the counterparty's node id for NextHop.

I tried the same thing that we do for UpdateFailed below this code chunk:

```
let mut channel_lock = self.channel_state.lock().unwrap();
let channel_state = &mut *channel_lock;
let by_id = &mut channel_state.by_id;


if let hash_map::Entry::Occupied(chan_entry) = by_id.entry(funding_outpoint.to_channel_id()) {
	let chan = chan_entry.remove_entry().1;
	self.fail_htlc_backwards_internal(self.channel_state.lock().unwrap(), htlc_update.source, &htlc_update.payment_hash, HTLCFailReason::Reason { failure_code: 0x4000 | 8, data: Vec::new() }, HTLCDestination::NextHopChannel { node_id: chan.get_counterparty_node_id(), channel_id: funding_outpoint.to_channel_id() });
}
```

however, this makes `test_fail_htlc_on_broadcast_after_claim` fail with the following error: `index out of bounds: the len is 0 but the index is 0`

which I presume is because the entry for `funding_outpoint.to_channel_id` does not exist.

any suggestions here for alternative ways to try and get the node id?

Reply:
> Oops, sorry I missed this somehow.
> 
> Grrrrr, right, that's annoying, we don't have any way to get that info here - the channel has closed so `ChannelManager` has no idea what it is, and `ChannelMonitor` never knew it to begin with. The only solution here is to give `ChannelMonitor` some knowledge of the counterparty node id, pipe it back through here, and make the counterparty node id optional in the event.
> 
> I think we should probably at least try to get that part done in a separate PR and slip it into the next release, it will end up being quite useful for the parallel channel access PR as well.

@TheBlueMatt Could you clarify what you're thinking here? I may have mistakenly thought this was similar to adding `ChannelMonitorUpdateStep::ShutdownScript` when `ChannelManager` learns of the shutdown script later and needs to communicate it to `ChannelMonitor`. But after re-reading your comment, this may not be an exact parallel to that, IIUC.